File: app/src/main/java/com/example/administrator/itemdecoration/bean/GroupInfo.java
package com.example.administrator.itemdecoration.bean;

/**
 * Created by lengwei on 2017/11/9.
 *
 * @Description
 */
public class GroupInfo {

    private String mGroupId;
    private String mTitle;
    private int mPosition;
    private int groupSize;

    public int getGroupSize() {
        return groupSize;
    }

    public void setGroupSize(int groupSize) {
        this.groupSize = groupSize;
    }

    public String getGroupId() {
        return mGroupId;
    }

    public void setGroupId(String groupId) {
        mGroupId = groupId;
    }

    public String getTitle() {
        return mTitle;
    }

    public void setTitle(String title) {
        mTitle = title;
    }

    public int getPosition() {
        return mPosition;
    }

    public void setPosition(int position) {
        mPosition = position;
    }

    public boolean isGroupFirstPosition(){
        return mPosition == 0;
    }

    public boolean isGroupLastPosition(){
        return mPosition == (groupSize -1) && mPosition >= 0;
    }
}

File: app/src/main/java/com/example/administrator/itemdecoration/bean/GroupInfo.kt
package com.example.administrator.itemdecoration.bean

/**
 * Created by lengwei on 2017/11/9.
 *
 * @Description
 */
class GroupInfo {

    var groupId: String? = null
    var title: String? = null
    var position: Int = 0
    var groupSize: Int = 0

    val isGroupFirstPosition: Boolean
        get() = position == 0

    val isGroupLastPosition: Boolean
        get() = position == groupSize - 1 && position >= 0
}
